File: ts/example/src/complex_user.ts
import {ExampleBothCommandAndQuery, ExampleCallback, ExampleManyMethods} from "./complex";



export class ActualBothCommandAndQueryUser {
  constructor(private cmdAndQuery: ExampleBothCommandAndQuery) {}


  doSomething(): number {
    // Do something.
    return this.cmdAndQuery.doSomething(true, 123);
  }
}


export class ActualManyMethodsUser {
  constructor(private manyMethods: ExampleManyMethods) {}

  doSomething(): number {
    this.manyMethods.foo(123);
    const i = this.manyMethods.bar();
    return this.manyMethods.baz(i);
  }
}


export class ActualCallbackUser {
  public isCalled = false;

  constructor(private callback: ExampleCallback) {}


  doSomething() {
    this.callback.doSomethingAsync(() => this.isCalled = true);
  }
}

File: swift/example/Sources/VanillaMockExample/ComplexUser.swift
class ActualBothCommandAndQueryUser {
    private let cmdAndQuery: ExampleBothCommandAndQueryProtocol


    init(cmdAndQuery: ExampleBothCommandAndQueryProtocol) {
        self.cmdAndQuery = cmdAndQuery
    }


    func doSomething() -> Int {
        self.cmdAndQuery.doSomething(foo: true, bar: 123)
    }
}



class ActualManyMethodsUser {
    private let manyMethods: ExampleManyMethodsProtocol


    init(manyMethods: ExampleManyMethodsProtocol) {
        self.manyMethods = manyMethods
    }


    func doSomething() -> Int {
        self.manyMethods.foo(123)
        let i = self.manyMethods.bar()
        return self.manyMethods.baz(i)
    }
}



class ActualCallbackUser {
    private let callback: ExampleCallbackProtocol
    private(set) var isCalled = false


    init(_ callback: ExampleCallbackProtocol) {
        self.callback = callback
    }


    func doSomething() {
        self.callback.doSomethingAsync { _ in self.isCalled = true }
    }
}